pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -515,6 +515,21 @@ func (i *Instances) NodeAddresses(nodeName k8stypes.NodeName) ([]v1.NodeAddress,
 			)
 		}
 	}
+
+	// ensure that the "primary" IP address of the machine is always included, since the above
+	// guest.net call is limited to 16 unsorted interfaces and the e.g. eth0 iface may not have
+	// been included.
+	err = getVirtualMachineManagedObjectReference(ctx, i.client, vm, "guest.ipAddress", &mvm)
+	if err != nil {
+		return nil, err
+	}
+	v1.AddToNodeAddresses(&addrs,
+		v1.NodeAddress{
+			Type:    v1.NodeInternalIP,
```

Comment:
That's a good question. I'm open to suggestions :)

I think I may have spent too much time in OpenStack. I think of the tenant network as internal, and a floating IP (or Elastic IP in AWS) as an external IP.